### PR TITLE
Update tests to run on Maven 3.9

### PR DESCRIPTION
**Pull request by willmostly**
Tests that passed on 3.6 are failing on 3.9. It looks like this is potentially due to stricter handling of NPEs. These changes define methods in the stub classes to prevent NPEs in the authentication layer.

### CODE REVIEW

**Review by vishalya — COMMENTED** [review]
Let's wait for this one, until we get all the tests running.

**Comment by mosabua** [comment]
We should move away from testng to junit and junit jupyter like Trino

**Review by andythsu — APPROVED** [review]

**Comment by mosabua** [comment]
Do we want to merge this and migrate back to Junit later ? Seems like we could just use junit now .. or is there some obstacle @willmostly ?

**Comment by willmostly** [comment]
Updated to junit 5, @vishalya and I are going to look into the test logic to see why the Ldap tests are getting connection timeouts

File: gateway-ha/pom.xml [hunk 1]
```diff
@@ -148,28 +148,46 @@
...
             <version>2.24.0</version>
             <scope>test</scope>
         </dependency>
+
         <dependency>
             <groupId>org.mockito</groupId>
-            <artifactId>mockito-all</artifactId>
-            <version>1.10.19</version>
+            <artifactId>mockito-core</artifactId>
+            <version>5.3.1</version>
```
Comment (mosabua) [inline gateway-ha/pom.xml:173]: Wow .. thats a BIG jump

**Comment by vishalya** [comment]
@willmostly - I have that test working with some minor changes on top of yours, https://github.com/vishalya/trino-gateway/commits/working_test

File: gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbFilter.java [hunk 2]
```diff
@@ -44,12 +46,23 @@ public void setup() throws Exception {
...
 
     // Set authorization manager with membership
     authorizationManager = Mockito.mock(AuthorizationManager.class);
     Mockito
         .when(authorizationManager.searchMemberOf(USER))
         .thenReturn(MEMBER_OF);
+    Mockito
+            .when(authorizationManager.getPrivileges(USER))
+            .thenReturn(MEMBER_OF);
+    /*
```
Comment (vishalya) [inline gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbFilter.java:59]: yes, you can get rid of this comment, you additional mocks are needed.

File: gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbFilter.java [hunk 3]
```diff
@@ -126,38 +141,41 @@ public void testSuccessfulHeaderAuthentication() throws Exception {
...
     // Checks authorization for authenticated principal
-    Assert.assertTrue(secContextCaptor.getValue().isUserInRole("USER"));
-    Assert.assertTrue(secContextCaptor.getValue().isUserInRole("ADMIN"));
+    assertTrue(secContextCaptor.getValue().isUserInRole("USER"));
+    assertTrue(secContextCaptor.getValue().isUserInRole("ADMIN"));
 
   }
 
-  @Test(expected = WebApplicationException.class)
+  @Test
```
Comment (vishalya) [inline gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbFilter.java:153]: how does the Junit 5 handle the expected exception?

Comment (willmostly) [reply gateway-ha/src/test/java/io/trino/gateway/ha/security/TestLbFilter.java:153]: In Junit 5 you wrap the method in an `assetThrows`, I used a lambda to do this here

**Review by vishalya — APPROVED** [review]

**Review by mosabua — APPROVED** [review]
Looks good now. Code style and further migration to junit will be done in follow up PRs.

**Pull request merged**